### DISCUSSION TITLE
gatsby-transformer-remark's peer dependency sanitize-html has vulnerabilities

### DISCUSSION THREAD

**Discussion by msaahil** [gatsbyjs/gatsby#34919] [Help]
### Security Vulnerability
The latest version of **gatsby-transformer-remark** uses **sanitize-html** version _**1.27.5**_.
And this version has a critical vulnerability of **Arbitrary Code Execution** according to a report by **[Snyk](https://snyk.io/)**.
sanitize-html needs to be upgraded to a **non-vulnerable** version.

Can we quickly resolve this ?
<img width="1139" alt="Screenshot 2022-02-24 at 1 07 38 PM" src="https://user-images.githubusercontent.com/100341082/155479841-583192d3-7053-4493-a9bf-5191b713949a.png">

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

While those audit messages might seem scary they are almost always irrelevant in the context of Gatsby as it's a build tool. So there is nothing to "fix" as the warning is not relevant.

Gatsby creates static assets and runs everything at build time and not during runtime. `npm audit` is designed for runtime / Node apps so it flags issues that can occur there. This means that almost every "vulnerability" report we receive are false positives. While in principle Gatsby can also have vulnerabilities you need to make sure that it's relevant to Gatsby before reporting it. For example a "Regex DDOS attack" can never be a real vulnerability for a development-time tool. If you want to override/update a transitive dependency you can use [yarn resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/).

You can also read [npm audit: Broken by Design](https://overreacted.io/npm-audit-broken-by-design/) to learn more about this.

If you know that a vulnerability affects Gatsby because you understand what the vulnerability is, please report it [here](https://github.com/gatsbyjs/gatsby/blob/master/SECURITY.md). Thanks!

**Reply by rriski** [reply]
Not updating the `sanitize-html` dependency in `gatsby-transformer-remark` is also leading to build issues for me:


```bash
failed Building development bundle - 14.821s
ERROR in ./.cache/blank.css
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
Error: PostCSS plugin postcss-flexbugs-fixes requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
```

The current `sanitize-html` version is specified as 1.27.5 which in turn depends on `postcss` 7.0.27 breaking the build.

@LekoArts I see there was a previous PR that got declined fixing this issue: https://github.com/gatsbyjs/gatsby/pull/31552 so I did not open a new one.
